src/main.rs

```diff
@@ -331,6 +338,9 @@ struct MainHandler {
 impl Handler for MainHandler {
     fn handle(&self, req: &mut Request) -> IronResult<Response> {
         let mut fs_path = self.root.clone();
+        if let Some(url) = &self.redirect_to {
+          return Ok(Response::with((status::PermanentRedirect, Redirect(url.clone()))));
```

Comment:
Please use `cargo fmt` to format the code.

Reply:
all set